src/engine/MaxConstraint.cpp

```diff
@@ -408,28 +452,94 @@ List<PiecewiseLinearCaseSplit> MaxConstraint::getCaseSplits() const
         splits.append( getSplit( _f ) );
     }
 
+    if ( _eliminatedVariables ) { // todo check with GuyK (I didn't add tightenings)
+        PiecewiseLinearCaseSplit phaseOfEliminatedIsMax;
+        // TODO - GuyA - move up to the case when _f exists only?
+        // f = maxEliminated
+        Equation maxEquation(Equation::EQ);
+        maxEquation.setScalar(_maxValueOfEliminated);
+        maxEquation.addAddend(1, _f);
+        phaseOfEliminatedIsMax.addEquation(maxEquation);
+
+        for (unsigned element : _elements)
+        {
+            if ( _elements.exists( _f ) && element == _f )
+                continue;
+
+            Equation ltEquation(Equation::LE);
+
+            // element <= maxEliminated
+            ltEquation.setScalar(_maxValueOfEliminated);
+            ltEquation.addAddend(1, element);
+            phaseOfEliminatedIsMax.addEquation(ltEquation);
+        }
+        splits.append( phaseOfEliminatedIsMax );
+    }
+
     return splits;
 }
 
 bool MaxConstraint::phaseFixed() const
-{
-    return _elements.size() == 1 || _elements.exists( _f );
+{ // todo updated - Guy A
+//    return _elements.exists( _f ) || _elements.size() == 1;
+    if ( _elements.exists( _f ) )
+        return true;
+    if ( _elements.size() == 1 )
+    {
+        if (!_eliminatedVariables)
+            return true;
+
+        unsigned singleVarLeft = *_elements.begin();
+        if ( _lowerBounds.exists( singleVarLeft ) && FloatUtils::gte( _lowerBounds[singleVarLeft], _maxValueOfEliminated ) )
+        {
+            return true;
+        }
+
+        if ( _upperBounds.exists( singleVarLeft ) && FloatUtils::lte( _upperBounds[singleVarLeft], _maxValueOfEliminated ) )
+        {
+            return true;
+        }
+
+    }
+    return false;
 }
 
-PiecewiseLinearCaseSplit MaxConstraint::getValidCaseSplit() const
-{
-    ASSERT( phaseFixed() );
-    if ( !_elements.exists( _f ) )
-        return getSplit( *( _elements.begin() ) );
-    else
-    {
+PiecewiseLinearCaseSplit MaxConstraint::getValidCaseSplit() const {
+    ASSERT(phaseFixed());
+    if ( !_elements.exists(_f) ) {
+        ASSERT(_elements.size() == 1);
+
+        unsigned singleVariableLeft = *_elements.begin();
+
+        if ( !_eliminatedVariables )
+            return getSplit(singleVariableLeft);
+
+        // eliminated at least one variable
+        // max = b = singleVariableLeft
+        if (_lowerBounds.exists(singleVariableLeft) &&
+            FloatUtils::gte(_lowerBounds[singleVariableLeft], _maxValueOfEliminated))
+            return getSplit(singleVariableLeft);
+
+        // max = maxValueEliminated
+        // maxEliminated - f = 0
+        PiecewiseLinearCaseSplit phaseOfEliminatedIsMax;
+        Equation maxEquation(Equation::EQ);
```

Comment:
Just set f's bounds to the eliminated var

Reply:
done